CHANGELOG.md

```diff
@@ -6,8 +6,18 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+### Added
+- E-mail address revalidation. At automated mailings (`irma keyshare tasks`) informing users their account is about to expire (temporary) invalid e-mail addresses are withheld from processing for 5 days, preventing cluttering of the processing queue
+
+**Note:** E-mail address revalidation requires a change in the database schema. In order to enable this feature please add the `revalidate_on` column of type `bigint` to the `irma.emails` table. See the [schema](https://github.com/privacybydesign/irmago/tree/master/server/keyshare/schema.sql#L50) file.
+
 ### Changed
 - Use separate application user in Dockerfile for entrypoint
+As part of e-mail address revalidation:
+- `VerifyMXRecord` incorporates a check to see if there is an active network connection
+- MyIrma server: `/user` returns an additional field `revalidate_in_progress` in the JSON response body, indicating whether the e-mail address is being revalidated or not 
+- MyIrma server: `/user/delete` and `/email/remove` return a 400 status code if one or more e-mail addresses of the user are invalid
+- MyIrma server: `/email/remove` returns a 400 status code if one or more e-mail addresses of the user are invalid
```

Comment:
It's not really something the user can change at this point anymore. Wouldn't a 5xx error make more sense in this case then?

Reply:
I could not find a fitting 5xx status code. If we'd use the general 500 status code we're not able to distinguish between a internal server error and invalid email addresses.

You can see the 400 error behavior on irma_keyshare_webclient [here](https://github.com/privacybydesign/irma_keyshare_webclient/pull/37/files#diff-68cfdabf9fa2ce99fa19fdbe25785f2622056c22f88b1f846d3ac7ffffea0744R69) and [here](https://github.com/privacybydesign/irma_keyshare_webclient/pull/37/files#diff-68cfdabf9fa2ce99fa19fdbe25785f2622056c22f88b1f846d3ac7ffffea0744R93). 

It's not ideal I agree. Otherwise we could maybe go for a 422 Unprocessable Content?

Reply:
Yeah it's not ideal. Let's keep the 400 then.